nose/subfindhdf_gadget4_test.py

```diff
@@ -0,0 +1,317 @@
+import pynbody
+import numpy as np
+from itertools import chain
+import shutil
+from nose.tools import assert_equals, assert_almost_equal
+import h5py
+
+
+def setup():
+    global snap, halos, subhalos, htest
+    snap = pynbody.load('testdata/testL10N64/snapshot_000.hdf5')
+    halos = snap.halos()
+    subhalos = snap.halos(subs=True)
+    htest = Halos('testdata/testL10N64/', 0)
+
+
+def teardown():
+    global snap, halos, subhalos, htest
+    del snap, halos, subhalos, htest
+
+
+def test_catalogue():
+    _h_nogrp = snap.halos(grp_array=True)
+    _subh_nogrp = snap.halos(subs=True, grp_array=True)
+    for h in [halos, subhalos, _h_nogrp, _subh_nogrp]:
+        assert(isinstance(h, pynbody.halo.subfindhdf.Gadget4SubfindHDFCatalogue)), \
+            "Should be a Gadget4SubfindHDFCatalogue catalogue but instead it is a " + str(type(h))
+
+
+def test_header():
+    assert(halos.header == htest.loadHeader())
+
+
+def test_halo_properties():
+    r = htest.load()['halos']
+    hids = np.random.choice(range(len(halos)), 5)
+    for hid in hids:
+        for key in list(r.keys()):
+            if key in list(halos[hid].properties.keys()):
+                np.testing.assert_allclose(halos[hid].properties[key], r[key][hid])
+
+
+def test_subhalo_properties():
+    r = htest.load()['subhalos']
+    hids = np.random.choice(range(len(subhalos)), 5)
+    for hid in hids:
+        for key in list(r.keys()):
+            if key in list(subhalos[hid].properties.keys()):
+                np.testing.assert_allclose(subhalos[hid].properties[key], r[key][hid])
+
+
+def test_halo_loading() :
+    """ Check that halo loading works """
+    h = halos
+    # check that data loading for individual fof groups works
+    _ = halos[0]['pos']
+    _ = halos[1]['pos']
+    _ = halos[0]['mass'].sum()
+    _ = halos[1]['mass'].sum()
+    assert(len(halos[0]['iord']) == len(halos[0]) == htest.load()['halos']['GroupLen'][0])
+
+
+def test_particle_data() :
+    hids = np.random.choice(range(len(halos)), 5)
+    for hid in hids:
+        assert(np.allclose(halos[hid].dm['iord'], htest[hid]['iord']))
+
+
+import six
+from os.path import isfile
+import numpy as np
+import h5py
+
+
+class Halos:
```

Comment:
Would be useful to add a docstring here to explain what it is doing in the context of this test